Dockerfile

```diff
@@ -0,0 +1,155 @@
+
+# ------------------------------------------------------
+# 3.2) Build zlib from source
+# ------------------------------------------------------
+ENV ZLIB_VERSION=1.3.1
+WORKDIR /tmp
+RUN curl -LO "https://www.zlib.net/zlib-${ZLIB_VERSION}.tar.gz" \
+    && tar zxvf "zlib-${ZLIB_VERSION}.tar.gz" \
+    && cd "zlib-${ZLIB_VERSION}" \
+    && ./configure --prefix=/usr/local \
+    && make -j"$(nproc)" \
+    && make install \
+    && cd / && rm -rf /tmp/zlib*
+
+WORKDIR /tmp
+RUN git clone https://chromium.googlesource.com/breakpad/breakpad
+WORKDIR /tmp/breakpad/
+RUN git clone https://chromium.googlesource.com/linux-syscall-support
+RUN mkdir -p /tmp/breakpad/src/third_party/lss/ && mv linux-syscall-support/linux_syscall_support.h src/third_party/lss/
+WORKDIR /tmp/breakpad
+# Pull linux-syscall-support submodule
+RUN ./configure && make && \
+    make install
+# 'make install' is not provided by breakpad, so we do it manually:
+# mkdir -p /usr/local/include/breakpad && \
+# mkdir -p /usr/local/lib && \
+# cp -r src/* /usr/local/include/breakpad/ && \
+# find . -name '*.a' -exec cp {} /usr/local/lib/ \; && \
+# Cleanup
+# cd / && rm -rf /tmp/breakpad
+
+# --------------------------------------------------------
+# 4) Pull the Latest OpenTTD Source
+# --------------------------------------------------------
+WORKDIR /build
+RUN git clone --depth=1 --branch ${APPVER} https://github.com/OpenTTD/OpenTTD.git
+
+# --------------------------------------------------------
+# 5) Build OpenTTD (Release Mode)
+# --------------------------------------------------------
+WORKDIR /build/OpenTTD
+RUN mkdir build && cd build && \
+    cmake .. -DCMAKE_BUILD_TYPE=Release &&\
+    make -j"$(nproc)" && \
+    make install
+
+# --------------------------------------------------------
+# 6) Expose Ports (TCP & UDP)
+# --------------------------------------------------------
+EXPOSE 3979/tcp
+EXPOSE 3979/udp
+
+
+# ------------------------------------------------------
+# 1) Builder Stage: Compile OpenGFX
+# ------------------------------------------------------
+# FROM ubuntu:22.04 AS builder
+
+# Install system dependencies and tools for building OpenGFX
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    git \
+    python3 \
+    python3-pip \
+    make \
+    imagemagick \
+    optipng grfcodec\
+    && rm -rf /var/lib/apt/lists/*
+
+# Install NML (required for compiling OpenGFX)
+RUN pip3 install nml
+
+# Clone the OpenGFX repository
+WORKDIR /build
+RUN git clone https://github.com/OpenTTD/OpenGFX.git
+
+# Compile OpenGFX via 'make'
+WORKDIR /build/OpenGFX
+RUN make
+
+
+
+# --------------------------------------------------------
+# 7) Default Command (Dedicated Server)
+# --------------------------------------------------------
+# We use 'sh -c' so that environment variables are expanded at runtime.
+# -D = Dedicated (no GUI)
+# -x <year> = Start year
+# -n 0.0.0.0:<port> = Listen on all interfaces at <port>
```

Comment:
`-n 0.0.0.0:<port>` does not listen on all interfaces; it definitely excludes IPv6 interfaces. With `-n [::]:<port>`you'll listen to both IPv4 and IPv6.